auth-service/src/main/java/com/shop/authservice/service/KafkaEventService.java

```
@@ -0,0 +1,194 @@
+
+    private final ObjectMapper objectMapper;
+    private final UserRepository userRepository;
+    private final KafkaTemplate<String, Object> kafkaTemplate;
+    private final Map<String, CompletableFuture<Boolean>> pendingCartChecks = new ConcurrentHashMap<>();
+    private final Map<String, CompletableFuture<List<UserCustomerDto>>> customersInfoRequests = new ConcurrentHashMap<>();
+    private final OutboxEventRepository outboxEventRepository;
+
+    public void sendRegistrationEvent(String email, Activation activation) {
+        try {
+            UserDataOperationEvent register = new UserDataOperationEvent(
+                    email,
+                    activation.getActivationCode(),
+                    activation.getExpiresAt(),
+                    LocalDateTime.now()
+            );
+
+            String json = objectMapper.writeValueAsString(register);
+            OutboxEvent event = new OutboxEvent(
+                    "user-registration-events",
+                    json,
+                    register.getClass().getName()
+            );
+            outboxEventRepository.save(event);
+        } catch (JsonProcessingException e) {
+            throw new RuntimeException(e);
+        }
+
+    }
+
+    public void sendPasswordResetEvent(String email, Activation activation) {
+        try {
+            UserDataOperationEvent register = new UserDataOperationEvent(
+                    email,
+                    activation.getActivationCode(),
+                    activation.getExpiresAt(),
+                    LocalDateTime.now()
+            );
+            String json = objectMapper.writeValueAsString(register);
+            OutboxEvent event = new OutboxEvent(
+                    "user-password-reset-events",
+                    json,
+                    register.getClass().getName()
+            );
+
+            outboxEventRepository.save(event);
+        } catch (JsonProcessingException e) {
+            throw new RuntimeException(e);
```

It's better to throw a custom exception, but it's a minor thing